Спринт №10

LasvegasandFlorida opened the pull request.

Commit:
Первый коммит dev

Commit:
Доработка dev

Commit:
Работа над спринтом №9

Commit:
Работа над спринтом №9 (ДЗ 2)

Commit:
Работа над спринтом №9 (ДЗ 4)

Commit:
- Добавлено наследование кнопок на "Главном экране"
- Упращен код перехода на экран "Поиска"
- Упращен код кнопки отправки письма на экране "Настройки"
- Код очистки строки поиска на экране "Поиска" вынесен в отдельную функцию

Commit:
- Добавлен RecyclerView для отображения треков
- RecyclerView заполнен мок-объектами

faritowich commented on app/src/main/java/com/example/playlistmakerprod/SearchActivity.kt:
```diff
@@ -0,0 +1,124 @@
+        return tracks.size
+    }
+
+}
```
Лучше разнести все эти классы по отдельным файлам. В проекте, где 3 класса находятся в одном файле, очень тяжело ориентироваться - на панели слева в дереве файлов показан только SearhActivity

faritowich commented on app/src/main/java/com/example/playlistmakerprod/SearchActivity.kt:
```diff
@@ -0,0 +1,124 @@
+        songName.text = model.trackName
+        songAuthor.text = model.artistName
+        songLength.text = model.trackTime
+        Glide.with(itemView).load(model.artworkUrl100).centerCrop().transform(RoundedCorners(2)).placeholder(R.drawable.placeholder).into(songImage)
```
Лучше числа, смысл которых сходу не понятен человеку, который впервые видит этот код, выносить в именованные переменные или константы (я про 2). На эту тему можно почитать, например, [эту статью](https://ru.hexlet.io/blog/posts/magic-numbers)

faritowich commented on app/src/main/java/com/example/playlistmakerprod/SearchActivity.kt:
```diff
@@ -0,0 +1,124 @@
+
+    override fun getItemCount(): Int {
+        return tracks.size
+    }
```
Можно сократить до `override fun getItemCount() = tracks.size`

faritowich commented on app/src/main/res/layout/songs_view.xml:
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8"?>
+    <LinearLayout
```
Немного поплыло форматирование. Есть хороший хоткей, можно нажать Alt + Ctrl + L и студия красиво отформатирует весь код на экране

faritowich commented on app/src/main/res/layout/songs_view.xml:
```diff
@@ -0,0 +1,63 @@
+            android:layout_weight="1"
+            android:layout_gravity="start"/>
+        <LinearLayout
+            android:layout_width="250dp"
```
Не стоит задавать конкретную ширину. Телефоны могут сильно отличаться по размерам, на разных телефонах будет выглядеть по разному. На маленьких устройства, где ширина сама по себе маленькая, часть может не влезть в экран

faritowich commented on app/src/main/res/layout/songs_view.xml:
```diff
@@ -0,0 +1,63 @@
+                android:textSize="@dimen/text_small"
+                android:textAppearance="@font/ys_display_regular"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"/>
```
Здесь и далее: для удобства визуального восприятия верстки в TextView удобно использовать атрибут `tools:text="Track Name"`
Он позволит указать текста в лейауте только для разработчика, при этом в приложении их не будет. Тогда не будет пустого TextView на превью, как сейчас, а будет видны тексты.
Чтобы использовать его, надо добавить в корневой лейаут строку:
    `xmlns:tools="http://schemas.android.com/tools"`

Commit:
- Добавлено получение applicationContext для экрана поиска
- Добавлена функция перевода dp в px для экрана поиска
- Изменен параметр функции RoundedCorners() для обложек треков
